### ISSUE TITLE
travis: workaround ssl issues when using py3 in trusty build environments

### ISSUE THREAD

**Issue by abn** [python-bugzilla/python-bugzilla#59]
The latest builds are failing due to an ssl exception [1]. This is isolated to trusty environments  with python 3.x.

```
Traceback (most recent call last):
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/urllib3/connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/urllib3/connectionpool.py", line 346, in _make_request
    self._validate_conn(conn)
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/urllib3/connectionpool.py", line 850, in _validate_conn
    conn.connect()
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/urllib3/connection.py", line 326, in connect
    ssl_context=context)
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/urllib3/util/ssl_.py", line 329, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/opt/python/3.3.6/lib/python3.3/ssl.py", line 231, in wrap_socket
    _context=self)
  File "/opt/python/3.3.6/lib/python3.3/ssl.py", line 335, in __init__
    raise x
  File "/opt/python/3.3.6/lib/python3.3/ssl.py", line 331, in __init__
    self.do_handshake()
  File "/opt/python/3.3.6/lib/python3.3/ssl.py", line 538, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:548)
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/requests/adapters.py", line 440, in send
    timeout=timeout
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/urllib3/connectionpool.py", line 639, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/urllib3/util/retry.py", line 388, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='bugzilla.mozilla.org', port=443): Max retries exceeded with url: /xmlrpc.cgi (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:548)'),))
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/travis/build/python-bugzilla/python-bugzilla/tests/ro_functional.py", line 133, in testVersion
    bz = Bugzilla("bugzilla.mozilla.org", use_creds=False)
  File "/home/travis/build/python-bugzilla/python-bugzilla/bugzilla/base.py", line 318, in __init__
    self.connect(url)
  File "/home/travis/build/python-bugzilla/python-bugzilla/bugzilla/base.py", line 561, in connect
    version = self._proxy.Bugzilla.version()["version"]
  File "/opt/python/3.3.6/lib/python3.3/xmlrpc/client.py", line 1090, in __call__
    return self.__send(self.__name, args)
  File "/opt/python/3.3.6/lib/python3.3/xmlrpc/client.py", line 1419, in __request
    verbose=self.__verbose
  File "/home/travis/build/python-bugzilla/python-bugzilla/bugzilla/transport.py", line 191, in request
    return self._request_helper(url, request_body)
  File "/home/travis/build/python-bugzilla/python-bugzilla/bugzilla/transport.py", line 154, in _request_helper
    url, data=request_body, **self.request_defaults)
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/requests/sessions.py", line 555, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/home/travis/build/python-bugzilla/python-bugzilla/.tox/py33/lib/python3.3/site-packages/requests/adapters.py", line 506, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='bugzilla.mozilla.org', port=443): Max retries exceeded with url: /xmlrpc.cgi (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:548)'),))
```

[1] https://travis-ci.org/python-bugzilla/python-bugzilla/builds/304713566

**Comment by crobinso** [comment]
@abn did you ever figure out how to work around this? Googling brings up a lot of hits but at first glance I couldn't find info specific to travis

**Comment by crobinso** [comment]
All the failures I've seen only seem to error for one call to mozilla's bugzilla, so we can just catch that specific case. Fixed in d89587f9ffaefc2dafab35296aef953c35e3e525
